src/OpenTelemetry/Metrics/Builder/MeterProviderBuilderExtensions.cs

```diff
@@ -313,14 +313,16 @@ public static MeterProviderBuilder ConfigureResource(this MeterProviderBuilder m
     /// </summary>
     /// <param name="meterProviderBuilder"><see cref="MeterProviderBuilder"/>.</param>
     /// <returns><see cref="MeterProvider"/>.</returns>
-    public static MeterProvider? Build(this MeterProviderBuilder meterProviderBuilder)
+    public static MeterProvider Build(this MeterProviderBuilder meterProviderBuilder)
     {
         if (meterProviderBuilder is MeterProviderBuilderBase meterProviderBuilderBase)
         {
             return meterProviderBuilderBase.InvokeBuild();
         }
 
-        return null;
+        OpenTelemetrySdkEventSource.Log.ProviderBuildCalledOnUnknownBuilder("Metrics", meterProviderBuilder?.GetType());
+
+        return new NoopMeterProvider();
```

Comment:
Any reason a noop instance is preferred instead of throwing exception here?

Reply:
I don't feel passionately either way about what it should do. A throw seemed a little more risky vs a no-op instance. But I doubt very much anyone has run into this case (having a custom builder type + calling the SDK build extension on it) so a throw is _probably_ fine. One note, for better or worse, there are a lot of SDK extensions that operate on top of the builder today that have this same sort of no-op-for-unknown-type behavior.

Reply:
If "throw is probably fine" let's go with throw.

Reply:
@alanwest Before I make changes, are you good with a throw here?

Reply:
I agree with "throw is probably fine". _Technically_ this is a breaking change, but I could also see us making an argument that it was a bug for this method to have been able to return null.

Reply:
Updated to throw.